scripts/communication.js

```diff
@@ -30,9 +30,10 @@ const responsesOnGreeting = [
 ];
 
 const handlerCommunication = (robot, queries, answers) => {
-  queries.forEach((query) => {
-    robot.hear(new RegExp(query, 'i'), response => response.send(response.random(answers)));
-  });
+  const matchKeys = queries.join('|');
+  const regexp = new RegExp(`\\b(${matchKeys})\\b`, 'i');
```

Comment:
It doesn't look like we have tests for this file so it's hard to say, but I'm quite cautious that it might not work when we handle what Mona says:

https://github.com/datarockets/mona/blob/7d3d887a233afd06ef51e5dc416b9bfc5342f0dd/scripts/communication.js#L40

Also from the demo it's not clear how it handles regexp items (we might consider to convert them into plain string this case):
https://github.com/datarockets/mona/blob/7d3d887a233afd06ef51e5dc416b9bfc5342f0dd/scripts/communication.js#L10

Reply:
@ezhlobo Good point 👍 Thanks. New demo:

![new-demo](https://user-images.githubusercontent.com/32716983/68934633-12d3ec00-07a8-11ea-85cc-5be854822e17.gif)